src/main/scala/com/mozilla/telemetry/learning/federated/FederatedLearningSearchOptimizerS3Sink.scala

```diff
@@ -0,0 +1,151 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.learning.federated
+
+import com.mozilla.telemetry.learning.federated.FederatedLearningSearchOptimizerConstants.{NumberOfFeatures, StartingLearningRate, StartingWeights}
+import com.mozilla.telemetry.streaming.FrecencyUpdateAggregate
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.Path
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.sql.execution.streaming.Sink
+import org.apache.spark.sql.sources.StreamSinkProvider
+import org.apache.spark.sql.streaming.OutputMode
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
+object FederatedLearningSearchOptimizerConstants {
+  // https://github.com/florian/federated-learning-addon/blob/master/addon/prefs.js#L26
```

Comment:
The default values are not in the addon anymore, so that link doesn't work anymore.
If you want a permalink to the current default values: https://dxr.mozilla.org/mozilla-central/rev/085cdfb90903d4985f0de1dc7786522d9fb45596/browser/app/profile/firefox.js#901